### PR TITLE
Added generalized phase retrieval method

### PR DESCRIPTION
**Pull request by mittoalb**
Added generalized phase retrieval method

ref: DOI 10.1088/2040-8986/abbab9

### CODE REVIEW

File: src/tomocupy/config.py [hunk 1]
```diff
@@ -242,6 +242,14 @@ def default_parameter(func, param):
         'default': 0,
         'type': float,
         'help': "Regularization parameter"},
+    'delta-beta': {
```
Comment (nikitinvv) [inline src/tomocupy/config.py]: could you call them as retrieve-phase-delta-beta and retrieve-phase-W? just to immediately understand that these pars  are needed for phase retrieval and can be easily found?

File: src/tomocupy/retrieve_phase.py [hunk 2]
```diff
@@ -195,6 +211,28 @@ def _reciprocal_grid(pixel_size, nx, ny):
...
+    -------
+    ndarray
+        Grid coordinates.
+    """
+    # Considering diffracting feature ~2*pixel size
+    # Sampling in reciprocal space.
+    indx = np.cos(_reciprocal_coord(pixel_size, nx)*2*PI*pixel_size)
+    indy = np.cos(_reciprocal_coord(pixel_size, ny)*2*PI*pixel_size)
+    
+    return np.add.outer(indx, indy)
```
Comment (nikitinvv) [inline src/tomocupy/retrieve_phase.py]: I believe you can do it like this if indx and indy are one-dimensional:
[indxx,indyy]=np.meshgrid(indy,indx)
indxx+indyy

**Review by nikitinvv — CHANGES_REQUESTED** [review]

File: src/tomocupy/config.py (hunk 1 above)
Comment (mittoalb) [reply src/tomocupy/config.py]: Done

File: src/tomocupy/retrieve_phase.py (hunk 2 above)
Comment (mittoalb) [reply src/tomocupy/retrieve_phase.py]: Modified both retrieval methods fully in cupy

**Review by nikitinvv — APPROVED** [review]
thanks @mittoalb

**Pull request merged**